### DISCUSSION TITLE
How to deploy a Quarto Interactive Document on a Shiny Server?

### DISCUSSION THREAD

**Discussion by cutterkom** [quarto-dev/quarto-cli#1997] [Q&A]
Hey, 

I want to deploy a Quarto Interactive Document on a Shiny Server in order to show it to the world. I expected it to work out of the box: Upload and ready to go. 

But I was wrong... What do I have to do besides uploading the `.qmd` to the Shiny Server? Uploading the `.html` breaks interactivity, of course.

My Quarto config looks like this:

```
---
title: "My title"
editor: source
format: 
  html:
    page-layout: full
    self-contained: true
server: shiny
---

```{r}
library(shiny)
...
...
```

I am aware that the Flexdashboard implementation is planned for Fall 2022, but simple interactive documents should be working as far I understand the post on https://quarto.org/docs/faq/rmarkdown.html.

How can I deploy a Quarto interactive document on a self-hosted Shiny Server?

**Comment by jjallaire** [comment]
You do indeed need to upload the `.html` file (as Shiny Server doesn't know how to render the `.qmd`). This does not break interactivity as the `.html` includes everything required to run the interactive document (including info on shiny dependencies, etc.). 

@jcheng5 Is there anything else special that needs to be done for `server: shiny` in Quarto interactive docs on Shiny Server?

**Comment by cutterkom** [comment]
Okay, I see. That works!
But only if it is **not** `self-contained: true`

Thanks!

**Comment by cutterkom** [comment]
Unfortunately, I have to unmark the answer. 

This is my `qmd` file: https://github.com/cutterkom/remove-na-lgbtiq-queer-knowledge-graph/blob/main/apps/remove-na-explore-surroundings/index.qmd

The URL is https://apps.katharinabrunner.de/remove-na-explore-surroundings/

The interactive parts don't work after an update: https://apps.katharinabrunner.de/remove-na-explore-surroundings/index.html
But it was working in a previous version. I did not change anything interactive, just text parts. Is it unstable? Am I missing something?

**Comment by jjallaire** [comment]
Best way to resolve this is to post an issue here: https://github.com/rstudio/shiny-server

**Comment by cutterkom** [comment]
I think it's a Quarto issue, not a Shiny Server one, since the local `html` doesn't work as expected as well (no interactive sections displayed)

**Comment by jjallaire** [comment]
The feature definitely works. Please see the documentation here: https://quarto.org/docs/interactive/shiny/. You have to call `quarto serve document.qmd` you can't just open the html in a browser.

**Reply by cutterkom** [reply]
Okay, weird. I opended an issue: https://github.com/rstudio/shiny-server/issues/530
Thanks for your help.
